curvlinops/inverse.py

```diff
@@ -627,6 +627,9 @@ def torch_matmat(self, M_torch: ParameterMatrixType) -> ParameterMatrixType:
         for mod_name, param_pos in self._A._mapping.items():
             # retrieve the inverses of the Kronecker factors from cache or invert them
             aaT_inv, ggT_inv = self._compute_or_get_cached_inverse(mod_name)
+            # cache the weight shape to ensure correct shapes are returned
+            if "weight" in param_pos:
+                weight_shape = M_torch[param_pos["weight"]].shape
```

Comment:
This is a stylistic comment: I would prefer to use `params` to reshape rather than the matrix supplied by the user, because this will reduce bugs due to user-supplied data:
```
# infer number of vectors onto which KFAC is applied
(V,) = {M.shape[0] for M in M_torch}
...
# look up parameter shapes of current block
w, b = self._params[w_pos], self._params[b_pos]
...
# explicitly reshape to un-flatten flattened dimensions when applying the Kronecker product
M_torch[w_pos] = (...).reshape(V, *w.shape)
M_torch[b_pos] = (...).reshape(V, *b.shape)
```
What do you think?

Reply:
We do explicitly check the shapes of the user supplied data [here](https://github.com/f-dangel/curvlinops/blob/main/curvlinops/kfac.py#L383-L401), so this shouldn't make a difference?